promql/parse_test.go

```diff
@@ -1590,48 +1589,29 @@ func TestParseExpressions(t *testing.T) {
 		expr, err := ParseExpr(test.input)
 
 		// Unexpected errors are always caused by a bug.
-		if err == errUnexpected {
-			t.Fatalf("unexpected error occurred")
-		}
-
-		if !test.fail && err != nil {
-			t.Errorf("error in input '%s'", test.input)
-			t.Fatalf("could not parse: %s", err)
-		}
+		testutil.Assert(t, err != errUnexpected, "unexpected error occurred")
 
-		if test.fail && err != nil {
+		if !test.fail {
+			testutil.Ok(t, err)
+			testutil.Equals(t, expr, test.expected, "error on input '%s'", test.input)
+		} else {
+			testutil.NotOk(t, err)
 			if !strings.Contains(err.Error(), test.errMsg) {
 				t.Errorf("unexpected error on input '%s'", test.input)
 				t.Fatalf("expected error to contain %q but got %q", test.errMsg, err)
 			}
```

Comment:
What would you/others think of adding a `testutil.Contains` function?

Reply:
As far as I understand, current mood about testutil is to keep it as small as possible. Moreover, I would remove this string comparison because it's essentially comparing errors and doing it via string comparison is fragile. I would create dedicated parser error types (maybe wrapped) and then matched those types in tests.

Reply:
In this case string comparisons are fine because it is also testing that a reasonable error message is going to be displayed to the user. For this PR how about putting the contains function into a `testutil.Assert` to remove the two `t.` lines?

Reply:
OK, done.